.github/workflows/ci-tests.yml

```diff
@@ -70,12 +89,13 @@ jobs:
 
       - name: Install resqpy
         run: |
-          python -m pip install --upgrade pip setuptools wheel
-          pip install -e ".[tests]"
+          pip install poetry
```

Comment:
It looks like this step uses a different method to install & configure Poetry than the "static analysis" one. For consistency, could we make them the same?

Given we will need to repeat several steps (install poetry, dependencies, caching etc), I think we could reduce duplication by using YAML anchors & aliases.

Reply:
Ah, I see that YAML anchors & aliases aren't actually supported in GitHub Actions. Shame!

An alternative is [Reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows), which seems to be essentially putting the common steps (Setup python, install poetry, install plugins, load deps from cache) into a separate yml workflow and then calling that as a step. Just a suggestion.